components/mdns/mdns_networking_lwip.c

```diff
@@ -21,6 +21,18 @@
 #include "mdns_networking.h"
 #include "esp_netif_net_stack.h"
 
+#if defined(CONFIG_MDNS_IPV4) && !LWIP_IPV4
+#error "Please enable IPv4 in LwIP to use IPv4 in mDNS component"
+#endif
+
+#if defined(CONFIG_MDNS_IPV6) && !LWIP_IPV6
+#error "Please enable IPv6 in LwIP to use IPv6 in mDNS component"
+#endif
+
+#if !defined(CONFIG_MDNS_IPV4) && !defined(CONFIG_MDNS_IPV6)
+#error "Neither IPv4 nor IPv6 is used in mDNS component"
+#endif
```

Comment:
We could get rid of these asserts if we find a better way of propagating which IP stack is enabled

Reply:
Removed